Provisioned Kafka clusters are detected / deleted, but serverless clusters are not

**Describe the bug**
Serverless Kafka clusters are not detected / deleted, only provisioned clusters get detected / deleted. 

**To Reproduce**
Steps to reproduce the behavior:

1. Go to the MSK console
2. Create a serverless Kafka cluster 
3. Create a provisioned Kafka cluster
4. Run the cleanup in dry run or destroy mode 
5. Check the execution log to see the Kafka cluster that was detected / deleted 

**Expected behavior**
Serverless Kafka clusters should be detected / deleted as well.

**AWS (please complete the following information):**

- Region: us-east-1

**Additional context**
Trying to troubleshoot the issue on my end and the issue appears to be related to the Kafka paginator that's being used: list_clusters which only detects provisioned clusters, but list_clusters_v2 detects both serverless and provisioned clusters. 

Haven't been able to get this working on my end though when changing the paginator.

Sorry for the delay, was out for a few. These are actually the changes I initially tested when trying to get the provisioned Kafka clusters to be detected / deleted as well. 

With these changes, now provisioned and serverless clusters are not shown in the execution log, the exec log is empty.

I am seeing this error in the app lambda logs:
`[ERROR] 'list_clusters_v2' (kafka_cleanup.py, clusters(), line 51)`

Yeah I did replace the **kafka:ListClusters** permission with **kafka:ListClustersV2**. Here are all the logs that mention Kafka:

```
[DEBUG] Started cleanup of Kafka Clusters. (kafka_cleanup.py, clusters(), line 33)

[ERROR] Could not list all Kafka Clusters. (kafka_cleanup.py, clusters(), line 50)

[ERROR] 'list_clusters_v2' (kafka_cleanup.py, clusters(), line 51)
```

I could not get this working. Really not sure what's causing the issue. Will take another look at it tomorrow.

**Here is what I tried**

- Included both **kafka:ListClusters** and **kafka:ListClustersV2** permissions
- Included only **kafka:ListClustersV2** permission
- Allowed all actions for all resources for the cleanup role
- Destroy all cleanup resources and redeployed

**What I found**

- No kafka clusters show up in the exec log when using **list_clusters_v2** in the kafka script, but when using **list_clusters** the provisioned clusters show up. 

I attached the output of when I run a**ws kafka list-clusters** and **aws kafka list-clusters-v2**:
[list-clusters.txt](https://github.com/servian/aws-auto-cleanup/files/8996843/list-clusters.txt)
[list-clusters-v2.txt](https://github.com/servian/aws-auto-cleanup/files/8996844/list-clusters-v2.txt)

--- a/app/serverless.yml
+++ b/app/serverless.yml
@@ -191,6 +191,7 @@ provider:
           Action:
             - kafka:DeleteCluster
             - kafka:ListClustersV2
+            - ec2:DeleteVpcEndpoints
           Resource: "*"
         - Effect: Allow
           Action:

